### PR TITLE
Add --keywords option to inspect command

### LINKED ISSUE #227: Extend the `inspect` command with `--keywords` to count JSON Schema keyword occurrences
Based on this quick experiment: https://github.com/sourcemeta-research/jsonschema-keywords

### PR DESCRIPTION
**Pull request by Nariman-Sayed**
Closes #227

Adds a `--keywords` flag to `inspect` that counts keyword occurrences in a schema grouped by vocabulary, based on the approach in https://github.com/sourcemeta-research/jsonschema-keywords

    jsonschema inspect schema.json --keywords

Uses SchemaIterator to walk the schema and count keywords per vocabulary. Made it conflict with --json since I wasn't sure how to combine keyword stats with the existing json output format, let me know if you had something else in mind.

Added tests for the normal case and the --json conflict, and updated the docs.

<a href="https://cubic.dev/pr/sourcemeta/jsonschema/pull/828?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: test/inspect/fail_keywords_with_json.sh
Comment (jviotti) [inline test/inspect/fail_keywords_with_json.sh:1]: Note that on `main` we largely replaced the shell tests with a new `.clitest` runner format. Can you adopt those?

File: test/inspect/pass_keywords.sh
Comment (jviotti) [inline test/inspect/pass_keywords.sh:1]: Let's have tests for this in conjunction with `--json`?

**Comment by jviotti** [comment]
Seems like CI is failing on a few OSes!

File: test/inspect/pass_keywords.sh
Comment (Nariman-Sayed) [reply test/inspect/pass_keywords.sh:1]: > Let's have tests for this in conjunction with `--json`?

I added fail_keywords_with_json.clitest for the --keywords + --json conflict case - let me know if you meant something else here.

File: src/command_inspect.cc [hunk 1]
```diff
@@ -152,13 +155,59 @@ auto print_frame(std::ostream &stream,
...
+
 auto sourcemeta::jsonschema::inspect(const sourcemeta::core::Options &options)
     -> void {
   if (options.positional().size() < 1) {
     throw PositionalArgumentError{"This command expects a path to a schema",
                                   "jsonschema inspect path/to/schema.json"};
   }
 
+  if (options.contains("keywords") && options.contains("json")) {
+    throw OptionConflictError{
```
Comment (jviotti) [inline src/command_inspect.cc:207]: @Nariman-Sayed Can we actually support `--json`? The entire command does support it (as almost every other command in the tool). In this case, we can present the same keyword summary, but print it as a JSON object or something like that.

File: test/inspect/pass_keywords.clitest
Comment (jviotti) [inline test/inspect/pass_keywords.clitest:1]: Would be nice to have a test showing the schema from standard input too?

Comment (jviotti) [reply test/inspect/pass_keywords.clitest:1]: Also maybe one that has unknown keywords. Also one where you pass a boolean schema, or a schema that is the empty object (in conjunction with `--default-dialect`). Overall thinking edge cases. Let's try to actually cover this feature in tests to try to surface any bug
